src/main/java/manufacturing/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,215 @@
+                        connection.prepareStatement(updateQuery);
+                PreparedStatement preparedStatementForInserting =
+                        connection.prepareStatement(insertQuery)) {
+            preparedStatementForDeletion.setLong(1, car.getId());
+            preparedStatementForDeletion.executeUpdate();
+            preparedStatementForUpdating.setString(1, car.getModel());
+            preparedStatementForUpdating.setLong(2, car.getManufacturer().getId());
+            preparedStatementForUpdating.setLong(3, car.getId());
+            preparedStatementForUpdating.executeUpdate();
+            preparedStatementForInserting.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                preparedStatementForInserting.setLong(2, driver.getId());
+                preparedStatementForInserting.executeUpdate();
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car in DB", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long carId) {
+        String query = "UPDATE cars "
+                + "SET deleted = TRUE "
+                + "WHERE car_id = ?";
+        int deletedID;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, carId);
+            deletedID = preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete drivers by id " + carId, e);
+        }
+        return deletedID > 0;
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.car_id, c.car_model, "
+                + "c.manufacturer_id, m.manufacturer_name, m.manufacturer_country "
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.manufacturer_id "
+                + "WHERE c.deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(createCar(resultSet, connection));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.car_id, cd.driver_id "
+                + "FROM cars c "
+                + "INNER JOIN cars_drivers cd "
+                + "ON c.car_id = cd.car_id "
+                + "WHERE driver_id = ? AND deleted = FALSE";
```

Comment:
we should also join manufacturers
and also drivers so that we could check that driver is not deleted

Reply:
done